sdk/metric/controller/push/push.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package push // import "go.opentelemetry.io/otel/sdk/metric/controller/push"
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/api/metric"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+)
+
+// Controller organizes a periodic push of metric data.
+type Controller struct {
+	sdk      *sdk.SDK
+	batcher  export.Batcher
+	exporter export.Exporter
+	ticker   *time.Ticker
+	wg       sync.WaitGroup
+	ch       chan struct{}
+}
+
+var _ metric.Provider = &Controller{}
+
+// New constructs a Controller, an implementation of metric.Provider,
+// using the provider batcher, exporter, period.  The batcher itself
+// is configured with aggregation policy selection.
+func New(batcher export.Batcher, exporter export.Exporter, period time.Duration) *Controller {
+	lencoder, _ := exporter.(export.LabelEncoder)
```

Comment:
```
// If the Exporter implements the export.LabelEncoder interface, the
// exporter will be used as the label encoder for the SDK itself,
// otherwise the SDK will be configured with the default label
// encoder.
```